### PR TITLE
Fix issue with power balancer on new-profile branch

### PR DESCRIPTION
**Pull request by cmcantalupo**
Fix bug in power balancer on new-profile branch and clean up the tests a bit.

### CODE REVIEW

File: integration/test/__main__.py [hunk 1]
```diff
@@ -56,11 +52,16 @@
 from test_launch_pthread import *
 from test_geopmagent import *
 from test_environment import *
-from test_power_governor import *
 from test_frequency_map import *
 from test_hint_time import *
 from test_progress import *
 
+if 'GEOPM_RUN_LONG_TESTS' in os.environ
```
Comment (bgeltz) [inline integration/test/__main__.py]: missing ```:``` at EOL

Was this tested?

File: integration/test/__main__.py [hunk 2]
```diff
@@ -56,11 +52,16 @@
...
+    from test_ee_timed_scaling_mix import *
+    from test_power_balancer import *
+    from test_power_governor import *
+    from test_scaling_region import *
+    from test_timed_scaling_region import *
```
Comment (bgeltz) [inline integration/test/__main__.py:64]:
```
else:
    print a nice warning message about skipped tests
```

File: test/SampleAggregatorTest.cpp [hunk 3]
```diff
@@ -339,9 +339,29 @@ TEST_F(SampleAggregatorTest, epoch_application_total)
 
     EXPECT_DOUBLE_EQ(2.0, m_agg->sample_region(M_SIGNAL_TIME, reg_normal));
     EXPECT_DOUBLE_EQ(2.0, m_agg->sample_region(M_SIGNAL_TIME, GEOPM_REGION_HASH_UNMARKED));
-    // First epoch observed at step == 2, app finished at step == 4.  4 - 2 = 2
```
Comment (bgeltz) [inline test/SampleAggregatorTest.cpp:342]: Some comments in the new code would really help in understanding what your changes are doing.  It took us a while to decipher this thing last night.

**Review by bgeltz — CHANGES_REQUESTED** [review]

File: integration/test/__main__.py (hunk 1 above)
Comment (cmcantalupo) [reply integration/test/__main__.py]: Obviously not :( thanks, will run main...

File: test/SampleAggregatorTest.cpp [hunk 4]
```diff
@@ -298,50 +298,83 @@ TEST_F(SampleAggregatorTest, epoch_application_total)
...
+    // and one second after for a total of two seconds
     EXPECT_DOUBLE_EQ(2.0, m_agg->sample_region(M_SIGNAL_TIME, reg_normal));
+    EXPECT_DOUBLE_EQ(1.0, m_agg->sample_region_last(M_SIGNAL_TIME, reg_normal));
+    // There have been two completed samples in region hash unmarked
+    // (this is the current region hash).
     EXPECT_DOUBLE_EQ(2.0, m_agg->sample_region(M_SIGNAL_TIME, GEOPM_REGION_HASH_UNMARKED));
-    // First epoch observed at step == 2, app finished at step == 4.  4 - 2 = 2
-    EXPECT_DOUBLE_EQ(2.0, m_agg->sample_epoch(M_SIGNAL_TIME));
+
+    /// Run through the same three region hashes with the epoch set to two
```
Comment (bgeltz) [inline test/SampleAggregatorTest.cpp]:
```suggestion
    // Run through the same three region hashes with the epoch set to two
```

**Review by bgeltz — APPROVED** [review]

**Pull request merged**